client/src/js/services/grid/TransactionService.js

```diff
@@ -303,24 +303,81 @@ function TransactionService($timeout, util, uiGridConstants, bhConstants, Notify
    * This method is used to check if a transaction is balanced by their record
    */
   function validateTransaction(entity) {
-    var transaction = entity.data.data;
-    var debit = 0,
-      credit = 0;
-    
-    transaction.forEach(function (row) {
-      debit += Number(row.debit_equiv);
-      credit += Number(row.credit_equiv);
+    var transaction = [];
+
+    // include transaction data
+    entity.data.data.forEach(function (row) {
+      transaction.push(row);
     });
 
-    var ERR_UNBALANCED_TXN = 'POSTING_JOURNAL.ERRORS.UNBALANCED_TRANSACTIONS';
+    // include new rows
+    entity.newRows.data.forEach(function (row) {
+      transaction.push(row);
+    });
 
-    // later in validateTransaction()
+    var numberOfLine = transaction.length;
     var error;
 
-    if (debit !== credit) {
-      error = ERR_UNBALANCED_TXN;
+    // If the transaction are single line transaction 
+    if(numberOfLine === 1){
+      error = 'POSTING_JOURNAL.ERRORS.SINGLE_LINE_TRANSACTION';
+    } else {
+      var debit = 0,
+        credit = 0,
+        initialDate,
+        accountNull = false,
+        dateNull =  false,
+        dateDifferent = false,
+        dateWrongPeriod = false;
+
+      if(transaction[0].trans_date){
+        initialDate = transaction[0].trans_date; 
+      }
+        
+
+      transaction.forEach(function (row) {
+        debit += Number(row.debit_equiv);
+        credit += Number(row.credit_equiv);
+
+
+        // Check if they are account number Null
+        if(!row.account_number){
```

Comment:
A lot of these checks you can simplify by removing the "if" and setting the error equal to the check.  LIke this:
```js
accountNull = !row.account_number;
dateNull = !row.trans_date;
dateDifference = Dates.util.str(row.trans_date) !== Dates.util.str(initialDate);
```
and so on.  This  might reduce your code a bit.

The comments are nice!